CRM/CiviAwards/Helper/CreateReviewFieldsValidator.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * Class CRM_CiviAwards_Helper_CreateReviewFieldsValidator.
+ */
+class CRM_CiviAwards_Helper_CreateReviewFieldsValidator {
```

Comment:
Since you have decided to move this to a class, it would have been better to make this generic a bit such that the constructor will accept `$fieldsLegalOptions` array and this could be used to validate arrays having similar structure. But let's do this later some other time since this has taken some time already.